specification/machinelearningservices/data-plane/Microsoft.MachineLearningServices/preview/2019-05-01/execution.json

```diff
@@ -0,0 +1,1103 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2019-05-01",
+    "title": "Execution Service API"
+  },
+  "schemes": [
+    "https"
+  ],
+  "paths": {
+    "/execution/v1.0/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/experiments/{experimentName}/localrun": {
+      "post": {
+        "tags": [
+          "Execution"
+        ],
+        "summary": "Start a run on a local machine.",
+        "description": "Starts an experiment run using the provided definition.json file to define the run.\r\n            The source code and configuration is defined in a zip archive in project.zip.",
+        "operationId": "Execution_StartLocalRun",
+        "consumes": [
+          "multipart/form-data"
+        ],
+        "produces": [
+          "application/octet-stream"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/workspaceNameParameter"
+          },
+          {
+            "name": "experimentName",
+            "in": "path",
+            "description": "The experiment name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "files",
+            "in": "formData",
+            "description": "Two files are required: 'project.zip':\r\n            A zip archive of a folder containing the source code to use for the run.\r\n            'definition.json': A text file that is a JSON-serialized run definition structure.",
+            "required": true,
+            "type": "array",
+            "items": {},
```

Comment:
What it the type of items?

Reply:
Hello @johanste,

This is the part I wanted your advise on how to resolve.
This particular API takes ICollection<IFormFile> files as input parameter.

Swashbuckle again doesn't convert the ICollection<IFormFile> type input param of operations to appropriate JSON structure gracefully.

To handle this issue, I have added builtin attributes [FromForm, SwaggerParameter(Required = true)] tags on the files parameter which is automatically converting it into appropriate JSON structure in swagger I suppose.

This API expects two files - 
1. .zip file(the whole project folder which needs to uploaded to cloud).
2. .json file(which details the configuration settings of the project)
Both the files are required.

Will you be able to suggest the expected JSON structure? 

I can add filters to generate swaggers accordingly

Reply:
@johanste Gentle reminder for your comment

Reply:
If it would have been possible, I would have suggested a different REST API. But since the API is already deployed, what you have there is probably the best you can do. You probably want to reach out to the ADX swagger reviewers internally to see if they have any other suggestions...

Reply:
If you have a suggestion for how to structure an API that requires a zip archive in addition to a JSON payload without:
* Requiring the service to inspect the archive
* Requiring clients to wait for an additional round-trip latency

I'm quite interested!

Reply:
However, we don't need this pattern for Execution_StartLocalRun -- just for Execution_StartRun.

Reply:
Couldnt come up any immediate fix. Aiming for better approach for the next release of swaggers with new API on service

Reply:
done